mava/components/building/datasets.py

```diff
@@ -114,7 +114,7 @@ def on_building_trainer_dataset(self, builder: SystemBuilder) -> None:
             postprocess=self.config.postprocess,
         )
 
-        builder.store.dataset_iterator = iter(dataset)
+        builder.store.dataset_iterator = dataset.as_numpy_iterator()
```

Comment:
I think we should keep both the dataset and dataset_iterator vars. That way we can test dataset props below.